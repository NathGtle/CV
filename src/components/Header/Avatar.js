import React from 'react'
import { StaticQuery, graphql } from 'gatsby'
import Img from 'gatsby-image'

const query = graphql`
  query {
    placeholderImage: file(relativePath: { eq: "avatar.jpg" }) {
      childImageSharp {
        fluid(maxWidth: 300) {
          ...GatsbyImageSharpFluid
        }
      }
    }
  }
`

export default () => (
  <StaticQuery
    query={query}
    render={data => <Img className="photo" fluid={data.placeholderImage.childImageSharp.fluid} />}
  />
)
