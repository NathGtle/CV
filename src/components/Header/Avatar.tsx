import React from 'react'
import PropTypes from 'prop-types'
import { useStaticQuery, graphql } from 'gatsby'
import Img from 'gatsby-image'

const Avatar = ({ alt }: AvatarInterface) => {
  const data = useStaticQuery(graphql`
    query {
      placeholderImage: file(relativePath: { eq: "avatar.jpg" }) {
        childImageSharp {
          fluid(maxWidth: 300) {
            ...GatsbyImageSharpFluid
          }
        }
      }
    }
  `)
  return <Img alt={alt} className="photo" fluid={data.placeholderImage.childImageSharp.fluid} />
}

Avatar.defaultProps = {
  alt: 'Kevin MARTIN',
}

Avatar.propTypes = {
  alt: PropTypes.string.isRequired,
}

export default Avatar
