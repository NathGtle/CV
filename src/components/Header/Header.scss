header {
  grid-area: logo;
  padding: $space-l;
}

.logo {
  color: #494949;
  font-size: 4rem;
  margin: 0;
}

.intro {
  margin: 0;
  color: $secondary-color;
  font-size: 2.5rem;
  text-transform: uppercase;
  padding-top: $space-m;
}

.photo {
  grid-area: photo;
  border-radius: 100px;
  height: 120px;
  width: 120px;
  display: none;

  @media screen and (min-width: $breakable-desktop) {
    display: flex;
    justify-self: center;
    align-self: center;
  }
}
