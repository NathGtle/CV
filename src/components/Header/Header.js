import React from 'react'
import { StaticQuery, graphql } from 'gatsby'

import AwesomeKevinMartinProfilePicture from './Avatar'

import './Header.scss'

const query = graphql`
  query {
    site {
      siteMetadata {
        lastname
        firstname
        job
      }
    }
  }
`

const Header = () => (
  <StaticQuery
    query={query}
    render={data => (
      <header className="header">
        <h1 className="logo">{`${data.site.siteMetadata.firstname} ${
          data.site.siteMetadata.lastname
        }`}</h1>
        <h2 className="intro">{data.site.siteMetadata.job}</h2>
        <AwesomeKevinMartinProfilePicture
          alt={`${data.site.siteMetadata.firstname} ${data.site.siteMetadata.lastname}`}
          className="photo"
        />
      </header>
    )}
  />
)

export default Header
