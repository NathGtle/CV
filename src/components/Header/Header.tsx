import React from 'react'
import { useStaticQuery, graphql } from 'gatsby'

import Avatar from './Avatar'

import './Header.scss'

const Header = () => {
  const data = useStaticQuery(graphql`
    query {
      site {
        siteMetadata {
          lastname
          firstname
          job
        }
      }
    }
  `)
  return (
    <React.Fragment>
      <header className="header">
        <h1 className="logo">{`${data.site.siteMetadata.firstname} ${data.site.siteMetadata.lastname}`}</h1>
        <h2 className="intro">{data.site.siteMetadata.job}</h2>
      </header>
      <Avatar alt={`${data.site.siteMetadata.firstname} ${data.site.siteMetadata.lastname}`} />
    </React.Fragment>
  )
}

export default Header
