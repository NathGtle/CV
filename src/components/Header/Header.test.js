import React from 'react'
import renderer from 'react-test-renderer'

import Header from './Header'
import { StaticQuery } from 'gatsby'

beforeEach(() => {
  StaticQuery.mockImplementationOnce(({ render }) =>
    render({
      site: {
        siteMetadata: {
          lastname: 'Martin',
          firstname: 'Kevin',
          job: 'GÃ©nie du mal',
        },
      },
    })
  )
})

jest.mock('./Avatar', () => 'mock-avatar')

describe('Header', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Header siteTitle="Default Starter" />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
