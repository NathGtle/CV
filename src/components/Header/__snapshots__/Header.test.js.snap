// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<header
  className="header"
>
  <h1
    className="logo"
  >
    Kevin Martin
  </h1>
  <h2
    className="intro"
  >
    GÃ©nie du mal
  </h2>
  <mock-avatar
    alt="Kevin Martin"
    className="photo"
  />
</header>
`;
